pkg/controllers/state/cluster.go

```diff
@@ -90,6 +103,14 @@ func (c *Cluster) Synced(ctx context.Context) (synced bool) {
 	// Set the metric to whatever the result of the Synced() call is
 	defer func() {
 		clusterStateSynced.Set(lo.Ternary[float64](synced, 1, 0))
+		if synced {
+			c.lastSynced = c.clock.Now()
+		}
+		// If we have been out of sync for a while, and we haven't logged about it recently, log it
+		if time.Since(c.lastSynced) > AbnormalClusterSyncDuration && c.cm.HasChanged("ClusterSynced", synced) {
+			logging.FromContext(ctx).Warnf("cluster state is out of sync since %v", time.Since(c.lastSynced))
```

Comment:
https://github.com/kubernetes/client-go/blob/aa7909e7d7c0661792ba21b9e882f3cd6ad0ce53/rest/request.go#L625 is how they log this in client-go for clientside throttling

Reply:
Makes sense. We really don't have a lot of warning logs in the codebase; I almost think that we shouldn't have them at all. What do you think about firing this as an error log and making sure that the sync duration is sufficiently long that a user would be convinced that this is actually an issue? 

Inevitably, the trouble with warning logs is that users end up treating them as Info (ignoring them because they don't actually mean anything) or they end up treating them as Error (because for them, the warning is always indicative of a problem). Warnings are basically an easy-out for a project to say "we don't know whether this can be a problem for you or not". Ideally, we can ship something that's always indicative of a problem

Reply:
yeah that makes sense, changed the severity for now